src/main/java/com/woowacamp/soolsool/core/member/domain/vo/MemberAddress.java

```diff
@@ -1,13 +1,35 @@
 package com.woowacamp.soolsool.core.member.domain.vo;
 
+import com.woowacamp.soolsool.global.exception.ShoppingException;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
-import lombok.RequiredArgsConstructor;
+import org.springframework.util.StringUtils;
 
 @Getter
 @EqualsAndHashCode
-@RequiredArgsConstructor
 public class MemberAddress {
 
+    private static final int MAX_LENGTH = 100;
+
     private final String address;
+
+    public MemberAddress(final String address) {
+        validateIsNotNullOrEmpty(address);
+        validateIsValidLength(address);
+
+        this.address = address;
+    }
+
+    private void validateIsValidLength(final String address) {
+        if (address.length() > MAX_LENGTH) {
+            throw new ShoppingException("회원 주소는 100자보다 길 수 없습니다.");
```

Comment:
예외 메시지의 100도 상수를 이용하면 좋을 것 같슴다!

```suggestion
            throw new ShoppingException("회원 주소는 " + MAX_LENGTH + "자보다 길 수 없습니다.");
```

Reply:
`ErrorCode`를 구현하는 `MemberErrorCode`를 정의했는데 상수를 어떻게 하면 좋을까요?

Reply:
음 그렇게 되면 상수를 넣기가 굉장히 어렵겠네요..

당장 떠오르는 방식은, 상수의 접근 제어자를 public으로 바꾸고 MemberErrorCode에서 바로 접근하는 것인데 접근 제어자를 활짝 여는 게 아무래도 마음에 들진 않네요 --;;

그래서 vo, MemberErrorCode에서 각각 관리하는 게 가장 낫지 않을까 싶은데 쩡규님 생각은 어떠신가여??

Reply:
좋은데요? 각각 관리 고고~

Reply:
는 enum에서 static field를 못 쓰네요 일단 매직넘보로 사용합시다